library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java

```diff
@@ -467,7 +467,8 @@ public void onSuccess(User user) {
                     public void onResponse(Call<QueryChannelsResponse> call, Response<QueryChannelsResponse> response) {
 
                         for (ChannelState channelState : response.body().getChannelStates()) {
-                            channelState.getLastMessage().setStatus(MessageStatus.RECEIVED);
+                            if (channelState.getLastMessage() != null)
+                                channelState.getLastMessage().setStatus(MessageStatus.RECEIVED);
```

Comment:
shouldn't we do this when we unmarshal JSON responses with messages? this approach sounds pretty fragile / prone to bugs (ie. we don't do this on `channel.query`)

Reply:
what's the reason for setting the message status here?